src/test/java/org/junit/internal/runners/statements/ExpectExceptionTest.java

```diff
@@ -0,0 +1,29 @@
+package org.junit.internal.runners.statements;
+
+import org.junit.Test;
+import org.junit.internal.AssumptionViolatedException;
+import org.junit.runners.model.Statement;
+
+import static org.junit.Assert.*;
+
+public class ExpectExceptionTest {
+
+    @Test
+    public void givenWeAreExpectingAssumptionViolatedExceptionAndAStatementThrowingAssumptionViolatedException() {
+        ExpectException sut = new ExpectException(new StatementThrowingAssumptionViolatedException(), AssumptionViolatedException.class);
+
+        try {
+            sut.evaluate();
+            // then no exception should be thrown
+        } catch (Throwable e) {
+            fail("should not throw anything, but was thrown: " + e);
```

Comment:
No need to catch unexpected exceptions; if `evaluate` throws an exception then the test would fail.